src/Subscriptions.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import ApolloClient, { ApolloError } from 'apollo-client';
+import { DocumentNode } from 'graphql';
+import { ZenObservable } from 'zen-observable-ts';
+import { OperationVariables } from './types';
+
+const shallowEqual = require('fbjs/lib/shallowEqual');
+const invariant = require('invariant');
+
+export interface SubscriptionResult<TData = any> {
+  loading: boolean;
+  data?: TData;
+  error: ApolloError;
+}
+
+export interface SubscriptionProps {
+  query: DocumentNode;
+  variables?: OperationVariables;
+  children: (result: any) => React.ReactNode;
+}
+
+export interface SubscriptionState<TData = any> {
+  loading: boolean;
+  data?: TData;
+  error?: ApolloError;
+}
+
+class Subscription<TData = any> extends React.Component<
+  SubscriptionProps,
+  SubscriptionState<TData>
+> {
+  static contextTypes = {
+    client: PropTypes.object.isRequired,
+  };
+
+  private client: ApolloClient<any>;
+  //TODO: What is the type here?
+  private queryObservable: any;
```

Comment:
From `ApolloClient.d.ts` (your IDE should allow you to follow the declaration of `client.subscribe`
```js
import {Observable} from 'apollo-client/util/Observable'
Observable<any>
```

Reply:
Great thanks!